mlflow/recipes/steps/train.py

```diff
@@ -465,6 +479,11 @@ def _get_user_defined_estimator(self, X_train, y_train, validation_df, run, outp
         sys.path.append(self.recipe_root)
         estimator_fn = getattr(importlib.import_module(train_module_name), estimator_method_name)
         estimator_hardcoded_params = self.step_config["estimator_params"]
+
+        # if using rebalancing pass in original class weights to preserve original distribution
+        if self.using_rebalancing:
+            estimator_hardcoded_params["class_weight"] = self.original_class_weights
```

Comment:
Reading the `class_weight` definition, this line seems to be no-op since the `class_weights` are computed in exactly the same way in `compute_class_weight` with `balanced` mode.
https://scikit-learn.org/stable/modules/generated/sklearn.utils.class_weight.compute_class_weight.html
https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html

Reply:
I could be wrong, but I believe this implementation is slightly different. The original class distribution is not known to the estimator since rebalancing is done upstream.

Reply:
Yes. @bbarnes52 You are right. The weights have changed effectively at this point.